### DISCUSSION TITLE
Relay module GPIO is not working.

### DISCUSSION THREAD

**Discussion by aleksKK** [rern/rAudio#394] [Fixed Issues]
rAudio 20210603 Raspberry Pi Zero W Rev 1.1 + HiFiBerry DAC

Enabled the relay module using system page. The default physical pins are 11, 13, 15, 16. During operation, all pins are invariably in input mode. (tested by pullup resistor) 

What else needs to be done? 
Thank you.

**Comment by rern** [comment]
The output mode is set at run time.
I've been using it everyday myself.
Please elaborate how you actually tested it to be sure that it's not working.

**Reply by aleksKK** [reply]
Thanks rern for the answer.

The GPIO status was monitored with an oscilloscope. State change registration is guaranteed.
Before and after shutdown, the state of the pins is always 0. GPIO mode was determined by connecting 3.3V through a 2K resistor. Connecting pullup-resistor always changed the level from 0 to 1. So GPIO is in input mode.
I checked the GPIO with a line in config.txt : gpio = 17,22,23,27 = op, dl
Connecting the pullup-resistor did not change the state of the pin. So GPIO is OK.

Please inform us if you did anything bypassing the WEB interface.
Thanks.

**Reply by rern** [reply]
The oscilloscope might be too much for just on/off testing. I'm sure you've got some LEDs and Rs to setup a simple test. No need to add the line in `config.txt`.

https://user-images.githubusercontent.com/10094236/121793976-82cd8100-cc2e-11eb-9a28-7e846e659332.mp4

More info: [+R GPIO](https://github.com/rern/R_GPIO/blob/master/README.md)

**Reply by aleksKK** [reply]
I thought that the on-sequence is executed after the voltage and the off-sequence after the shutdown. This is my mistake. 
All works well from the special icon. Thanks rern. 
Is it possible to start an on-sequence by applying voltage (without an icon)? 
Thank you.
